eng/scripts/SignAndHash-MavenPackages.ps1

```diff
@@ -0,0 +1,177 @@
+#Requires -Version 7
+
+param(
+  [Parameter(Mandatory=$true)][string]$Path,
+  [Parameter(Mandatory=$true)][string]$DestinationPath,
+  [Parameter(Mandatory=$true)][string]$GPGExecutablePath,
+  [Parameter(Mandatory=$false)][AllowEmptyString()][string]$GroupIDFilter,
+  [Parameter(Mandatory=$false)][AllowEmptyString()][string]$ArtifactIDFilter
+)
+
+Set-StrictMode -Version 2.0
+
+$ErrorActionPreference = "Stop"
+
+. $PSScriptRoot\MavenPackaging.ps1
+
+# The Resolve-Path will normalize the path separators and throw if they don't exist.
+# This is necessary because, the yml passes in ${{parameters.BuildToolsPath}}/ which
+# on Windows means <drive>:\<BuildToolsPath>/<whatever>/<else>. Maven doesn't always
+# behave well with different separators in the path.
+$Path = Resolve-Path $Path
+$GPGExecutablePath = Resolve-Path $GPGExecutablePath
+
+function Get-RandomRepositoryDirectory() {
+  Write-Host "Getting random repository directory."
+  $randomSubDirectoryName = [System.IO.Path]::GetRandomFileName()
+  $randomRepositoryDirectory = New-Item -Type Directory -Path $env:TEMP -Name $randomSubDirectoryName
+  Write-Host "Random repository directory is: $randomRepositoryDirectory"
+  return $randomRepositoryDirectory
+}
+
+function ConvertTo-DeploymentDetails($PackageDetail) {
+  $pomArtifact = $packageDetail.AssociatedArtifacts | Where-Object { ($null -eq $_.Classifier) -and ($_.Type -eq "pom") }
```

Comment:
nit: We should consider using `.Where({...}) over | Where-Object` as it will ensure you always end up with an array.